### PR TITLE
[DoctrineBridge] [DX] Update exception text in ManagerRegistry to avoid confusion.

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yesish 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | #29659   
| License       | MIT
| Doc PR        | 

Since the last PR was closed and the ticket is still open, taking it since it was already done by Nicolas in the comments.

### CODE REVIEW

**Comment by Simperfit** [comment]
Failure is not related

File: src/Symfony/Bridge/Doctrine/ManagerRegistry.php [hunk 1]
```diff
@@ -46,7 +46,7 @@ protected function resetService($name)
         $manager = $this->container->get($name);
 
         if (!$manager instanceof LazyLoadingInterface) {
-            throw new \LogicException(sprintf('Resetting a non-lazy manager service is not supported. Set the "%s" service as lazy and require "symfony/proxy-manager-bridge" in your composer.json file instead.', $name));
+            throw new \LogicException(sprintf('Resetting a non-lazy manager service is not supported. '.(interface_exists(LazyLoadingInterface::class) ? sprintf('Declare the "%s" service as lazy.', $name) : 'Try running "composer require symfony/proxy-manager-bridge".')));
```
Comment (jewome62) [inline src/Symfony/Bridge/Doctrine/ManagerRegistry.php]: you can remove the first sprintf, because not argument added

**Review by jewome62 — CHANGES_REQUESTED** [review]

**Comment by Simperfit** [comment]
Status: Needs Review

**Comment by nicolas-grekas** [comment]
Thank you @Simperfit.

**Comment by guilliamxavier** [comment]
@nicolas-grekas: Sorry but that just made the error message even *more* misleading in my case:

> Resetting a non-lazy manager service is not supported. Declare the "doctrine.orm.default_entity_manager" service as lazy.

but it is *already* lazy, says my `bin/console debug:container doctrine.orm.default_entity_manager`. The actual thing to do is `composer require symfony/proxy-manager-bridge` (then there is no error anymore), but that information is no longer in the error message.

The problem is that `ProxyManager\Proxy\LazyLoadingInterface` is defined not by `symfony/proxy-manager-bridge` but by `ocramius/proxy-manager`, and I have `ocramius/proxy-manager` installed _(because required by `doctrine/migrations` required by `doctrine/doctrine-migrations-bundle` required by `symfony/orm-pack` required in my `composer.json`)_ without having `symfony/proxy-manager-bridge` installed, so the interface exists but it is not sufficient.

Couldn't the code check that `symfony/proxy-manager-bridge` is actually installed (given it also requires `ocramius/proxy-manager`)?

**Comment by nicolas-grekas** [comment]
Nice suggestion, PR welcome :)

**Comment by guilliamxavier** [comment]
@nicolas-grekas: I'm not sure how to do. Should I replace `interface_exists(LazyLoadingInterface::class)` with `class_exists(RuntimeInstantiator::class)` (adding a `use Symfony\Bridge\ProxyManager\LazyProxy\Instantiator\RuntimeInstantiator;`)? Or maybe keep the first and append the second with `&&`? Or another approach entirely?
Also, PR on branch `4.3`?

**Comment by nicolas-grekas** [comment]
Please open an issue or a PR so that we can keep track of the discussion.

**Pull request merged**